apps/coscrad-frontend/src/components/commands/connections/connect-resources-with-note-form.tsx

```diff
@@ -0,0 +1,125 @@
+import { isNonEmptyString } from '@coscrad/validation-constraints';
+import {
+    Box,
+    Button,
+    FormGroup,
+    MenuItem,
+    Select,
+    Stack,
+    TextField,
+    Typography,
+} from '@mui/material';
+import { useState } from 'react';
+import { AggregateDetailContainer } from '../../higher-order-components/aggregate-detail-container';
+import { GlobalSearch } from '../../higher-order-components/global-search';
+import { fullViewCategorizablePresenterFactory } from '../../resources/factories/full-view-categorizable-presenter-factory';
+
+export interface ConnectResourcesWithNoteFormProps {
+    /**
+     * In the future, we can make this optional. For now the only use case
+     * is to bind `CONNECT_RESOURCES_WITH_NOTE` to a resource detail view, with
+     * the corresponding resource playing the role of the `from` member.
+     */
+    fromMemberCompositeIdentifier: ResourceCompositeIdentifier;
+    onSubmitForm: (fsa?: { type: string; payload: Record<string, unknown> }) => void;
+    bindProps: Record<string, unknown>;
+}
+
+export const ConnectResourcesWithNoteForm = ({
+    fromMemberCompositeIdentifier,
+    onSubmitForm,
+    bindProps,
+}: ConnectResourcesWithNoteFormProps) => {
+    const [text, setText] = useState<string>('');
+
+    const [languageCode, setLanguageCode] = useState<LanguageCode>(null);
+
+    const [toMemberCompositeIdentifier, setToMemberCompositeIdentifier] =
+        useState<ResourceCompositeIdentifier>();
+
+    const isFormComplete =
+        isNonEmptyString(text) &&
+        Object.values(LanguageCode).includes(languageCode) &&
+        isResourceType(toMemberCompositeIdentifier?.type) &&
+        isNonEmptyString(toMemberCompositeIdentifier?.id);
+
+    return (
+        <Box>
+            <Stack>
+                <Typography variant="h2">Connect Resources with Note</Typography>
+                <FormGroup>
+                    <TextField
+                        id="note_text"
+                        name="text"
+                        label="note text"
+                        value={text}
+                        onChange={(e) => {
+                            setText(e.target.value);
+                        }}
+                        multiline
+                        required
+                    />
+                </FormGroup>
+                <FormGroup>
+                    <Select
+                        id="note_languageCode"
+                        labelId="note_languageCode-label"
+                        value={languageCode || ''}
+                        label="language"
+                        name="language"
+                        placeholder="Choose Language"
+                        onChange={(e) => {
+                            setLanguageCode(e.target.value as LanguageCode);
+                        }}
+                        required
+                    >
+                        {Object.entries(LanguageCode).map(([languageLabel, languageCode]) => (
```

Comment:
I wonder if we should have a `LanguageCodeSelect` if we are going to build this here. Or is there some kind of `SelectFromEnum` helper?

Reply:
I suppose there's not a lot of value in doing this as we will eventually phase out these hard-wired forms which are a temporary hack.